scipy/spatial/transform/_rotation.pyx

```diff
@@ -363,9 +363,17 @@ cdef double[:, :] _compute_euler_from_quat(
 
     return angles
 
+
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cdef inline void _compose_quat_single( # calculate p * q into r
+cdef inline double _quat_magnitude(const double[:] q):
```

Comment:
Honestly, I would prefer not to have these additional changes in this PR.

If you think it's important to refactor usage of magnitude computations, you can easily do it in other PR.

Reply:
Sounds good, removed them and addressed your other comments.